tests/integration/targets/setup_cron/tasks/main.yml

```diff
@@ -25,10 +25,10 @@
   - name: install cron package
     package:
       name: '{{ cron_pkg }}'
-    when: cron_pkg | default(false, true)
+    when: (cron_pkg | default(false, true)) is truthy
```

Comment:
Why?

Reply:
If `cron_pkg` is a non-empty string, `cron_pkg | default(false, true)` is still a non-empty string, and Ansible will fail since it no longer accepts non-empty strings when there should be a bool.

Reply:
Does that extend to other types as well? Will it stop to recognize `0` as `false` in a `bool` context and `1` as `true`?

Reply:
I guess it depends on the context ;) In boolean conditions in playbooks (like `when` or `assert`/`that`) I think it will only accept proper booleans.